src/api/routes.ts

```diff
@@ -224,43 +224,71 @@ export function getTarball(req: auth.AuthRequest, res: express.Response) {
 }
 
 export function publishPackage(req: auth.AuthRequest, res: express.Response): void {
-  let jsonErrorResponse: any = { message: 'error saving package version' };
   let name: string = req.params.package;
   let authFile = getAuth();
   let user: any = auth.getUserByToken(res.locals.remote_user.token, authFile);
-  let organization: string = null;
-  let teams: any = null;
-  if (name[0] === '@') {
-    let splitName = name.split('/')[0].slice(1);
-    if (splitName !== user.name) {
-      organization = splitName;
-      teams = auth.getUserTeams(user.name, organization, authFile).map(t => {
-        return { team: t.name, permission: 'read-write' };
+  let request = req.headers.referer.split(' ');
+  if (request[0] === 'star') {
+    let username = user.name;
+    if (username in req.body.users) {
+      auth.starPackage(user.name, name, authFile).then((newAuthFile) => {
```

Comment:
I see what you mean... Well yeah, the same route is called when you want to publish a package or star it... I think I should rename this function into something similar like updatePackage...